Lib/test/test_typing.py

```diff
@@ -4391,6 +4391,21 @@ def blah():
 
         blah()
 
+    @patch("typing._overload_registry",
+        defaultdict(lambda: defaultdict(dict)))
+    def test_overload_on_compiled_functions(self):
+        # The registry starts out empty:
+        self.assertEqual(typing._overload_registry, {})
+
+        # This should just not fail:
+        overload(sum)
+        overload(print)
+
+        # No overloads are recorded (but, it still has a side-effect):
+        self.assertEqual(typing.get_overloads(sum), [])
+        self.assertEqual(typing.get_overloads(print), [])
+        self.assertNotEqual(typing._overload_registry, {})
```

Comment:
This last line feels like it's testing for incorrect behaviour. I don't think it's a massive _problem_ that `overload(sum)` has a side effect, but I still feel like, in an ideal world, it wouldn't have any side effects.

```suggestion
```

Reply:
I am fine with removing it, my motivation was to show that side-effect happens.